File: go/darts/darts.go
// Package Darts contains a method score to return the score of a dart at a given coordinate.
package darts

// Score - given two co-ordinates, a score is returned as an integer.
func Score(x, y float64) int {

	distance := x*x + y*y

	if distance <= 1 {
		return 10
	} else if distance <= 25 {
		return 5
	} else if distance <= 100 {
		return 1
	} else {
		return 0
	}

}

File: c/darts/src/darts.c
#include "darts.h"


uint8_t score(struct coordinate_t landing_position) {

    float position = (landing_position.x *landing_position.x) + (landing_position.y*landing_position.y);

    if (position <= 1) {
        return 10;
    } 
    else if (position <= 25) {
        return 5;
    }
    else if (position <= 100) {
        return 1;
    }
    else {
        return 0;
    }
}
